source/javascripts/services/_appService.js.erb

```diff
@@ -643,6 +627,51 @@
         return [];
       }
 
+      appService.detectRemoteAppConfigConflict = function (requestConfig) {
+        return $q.all([
+          BitriseYmlApi.formatYml(appService.appConfig),
+          requestService.getAppConfig(requestConfig),
+        ]).then(function ([yourFormattedConfig, remoteConfig]) {
+          appService.remoteAppConfig = normalizeAppConfig(remoteConfig);
+
+          return {
+            yourYaml: yourFormattedConfig,
+            baseYaml: BitriseYmlApi.toYml(appService.savedAppConfig),
+            remoteYaml: BitriseYmlApi.toYml(appService.remoteAppConfig),
+            hasConflict: !angular.equals(appService.remoteAppConfig, appService.savedAppConfig),
+          }
+        });
+      };
+
+      appService.detectRemoteAppConfigYMLConflict = function (requestConfig) {
+        return requestService.getAppConfigYML(requestConfig).then(function (remoteYaml) {
+          appService.remoteAppConfigYML = remoteYaml;
+
+          return {
+            yourYaml: appService.appConfigYML,
+            baseYaml: appService.savedAppConfigYML,
+            remoteYaml: remoteYaml,
+            hasConflict: remoteYaml !== appService.savedAppConfigYML,
```

Comment:
Do we plan to make this smarter? In this form, is not really conflict detection but simply change detection. We could have more suitable name for this like `hasChange` since the `hasConflict` name is slightly misleading I think.